org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/CompletionTests18.java

```diff
@@ -6955,4 +6955,31 @@ public static GH979List<String> newInstance() {
 					+ (R_DEFAULT + R_RESOLVED + R_INTERESTING + R_CASE + R_UNQUALIFIED + R_NON_RESTRICTED) + "}",
 			result);
 }
+
+public void testGH979_onAnonClassConstructorWith_expectOnlyAnonClassCompletion() throws JavaModelException {
+	this.workingCopies = new ICompilationUnit[1];
+	this.workingCopies[0] = getWorkingCopy("/Completion/src/GH979.java", """
+		import java.io.Serializable;
+		public class GH979 {
+				public GH979() {}
+
+				public void foo() {
+					Serializable run= new Serializable() {
```

Comment:
Is it possible to use a local type and not a JVM type?

Reply:
Just copied and merge the snippet for jdt.ui.text test. But good suggestion, now i changed it.